modules/core/shared/src/main/scala/io/circe/Json.scala

```diff
@@ -479,6 +479,8 @@ object Json {
   final val Null: Json = JNull
   final val True: Json = JBoolean(true)
   final val False: Json = JBoolean(false)
+  private[this] final val EmptyArray: Json = JArray(Vector.empty)
```

Comment:
private[this] will be removed in scala 3 at some point, so we should not use that
https://docs.scala-lang.org/scala3/reference/dropped-features/this-qualifier.html

Reply:
Sure, but it's still needed on Scala 2 for performance.

Reply:
I don't have any strong opinions about this, so I'll do whatever you want, but I think I used `private[this]` because I saw that being used in other places.

(I'm on holiday, so it's not very convenient for me to link to other usages)

Reply:
I'm finally back, so I can take a look at this.

However, now I'm confused, should I change this or not?

Reply:
I suppose we will put a pin in that for now. Once scala-3 drops this I guess we will have to split scala-2 and scala-3 for these cases.